tools/scripts/generic.mk

```diff
@@ -250,7 +253,14 @@ REL_SRCS = $(addprefix $(OBJECTS_DIR)/,$(call get_relative_path,$(SRCS)))
 OBJS = $(REL_SRCS:.c=.o)
 
 REL_ASM_SRCS = $(addprefix $(OBJECTS_DIR)/,$(call get_relative_path,$(ASM_SRCS)))
-ASM_OBJS = $(REL_ASM_SRCS:.S=.o)
+ASM_OBJS_s = $(REL_ASM_SRCS:.s=.o)
+ifneq ($(REL_ASM_SRCS),$(ASM_OBJS_s))
+	ASM_OBJS += $(ASM_OBJS_s)
+endif
+ASM_OBJS_S = $(REL_ASM_SRCS:.S=.o)
+ifneq ($(REL_ASM_SRCS),$(ASM_OBJS_S))
+	ASM_OBJS += $(ASM_OBJS_S)
+endif
```

Comment:
Will test this and include your change if it works, it's shorter ;)

Reply:
Although what do you do when you have a mix of .asm .s and .S suffixes ?

Reply:
It can be done using wildcard somehow but for aducm3029 there is just on .S file and I assume for stm32 there are only .s files so I think there is no need to treat that case for the moment

Reply:
```suggestion
ASM_SUFFIX = $(suffix $(REL_ASM_SRCS))
ASM_OBJS  = $(REL_ASM_SRCS:.s=.o)
ASM_OBJS  := $(ASM_OBJS:.S=.o)
```